src/cdk-new/create-rest-api.ts

```diff
@@ -0,0 +1,42 @@
+import type {RestApiBase, StageOptions} from 'aws-cdk-lib/aws-apigateway';
+import {
+  EndpointType,
+  RestApi,
+  SecurityPolicy,
+} from 'aws-cdk-lib/aws-apigateway';
+import type {ICertificate} from 'aws-cdk-lib/aws-certificatemanager';
+import type {Stack} from 'aws-cdk-lib/core';
+import {CfnOutput} from 'aws-cdk-lib/core';
+
+export interface RestApiInit {
+  readonly stack: Stack;
+  readonly certificate: ICertificate;
+  readonly restApiName: string;
+  readonly domainName: string;
+  readonly stageOptions: StageOptions;
+}
+
+export function createRestApi(init: RestApiInit): RestApiBase {
+  const {stack, certificate, restApiName, domainName, stageOptions} = init;
+
+  const restApi = new RestApi(stack, `RestApi`, {
+    endpointTypes: [EndpointType.REGIONAL],
```

Comment:
if you don't have an CDN before the stack, you might want to use edge optimized

Reply:
Yes possibly, but that has implications. For example, the certificate must be created in US-East-1. Means 2 stacks. Until now, AWS-simple has also only supported regional endpoints. I will not change this for now. If necessary, you can put a CloudFront in front of the API gateway. I would recommend that anyway.

Reply:
if a user provides his own certificate ARN he could safely choose EDGE then?